### PR TITLE
Implement initial version of documentation

### PR DESCRIPTION
**Pull request by giovannicimolin**
This PR introduces the initial version of a new documentation.
Check the current version at: https://giovannicimolin.github.io/react-native-iap/

The current version is basically a copy-and-paste of the current README, presented in a more categorized way and with a few wording fixes.

@hyochan Can you do a review pass on this? 

Related discussion: https://github.com/dooboolab/react-native-iap/discussions/1365#discussion-3397865

### CODE REVIEW

**Review by hyochan — CHANGES_REQUESTED** [review]
This just looks fabulous :trophy: 

Could you also handle below links?
image

As seen above, it'd be good to direct to `react-native-iap` issues (the actual link is not provided).

**Comment by giovannicimolin** [comment]
@hyochan Yup, I'll fix those links a little bit later in the week 👍

**Comment by hyochan** [comment]
> @hyochan Yup, I'll fix those links a little bit later in the week

You might also want to remove the content in the `readme` in our repo. The same thing doesn't have to belong twice.

**Comment by giovannicimolin** [comment]
@hyochan I've finished moving the docs to Docussaurus, fixed the links and cleaned up the README.md a bit.
Latest version is up here: https://giovannicimolin.github.io/react-native-iap/docs/intro

File: docs/docs/usage_instructions/retrieve_available.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  ],
+  android: [
+    'com.example.coins100'
+  ]
+});
+```
+
+To get a list of valid items, call `getProducts()`.
+
+You can do it in `componentDidMount()`, or another area as appropriate for you app.
```
Comment (hyochan) [inline docs/docs/usage_instructions/retrieve_available.md:24]: `you` => `your`. Already mistake from previous `readme` tough.

**Review by hyochan — APPROVED** [review]
This is more than great!!! :tada: 

We finally have the document!!

Please submit `$30` of expense in https://opencollective.com/react-native-iap for a small gift for you!

File: docs/docs/usage_instructions/retrieve_available.md (hunk 1 above)
Comment (giovannicimolin) [reply docs/docs/usage_instructions/retrieve_available.md:24]:
```suggestion
You can do it in `componentDidMount()`, or another area as appropriate for your app.
```

**Comment by giovannicimolin** [comment]
@hyochan 
>Please submit $30 of expense in https://opencollective.com/react-native-iap for a small gift for you!

This is an awesome surprise! Thank you for being supportive of open source contributions. 😄

**Comment by giovannicimolin** [comment]
@hyochan There are a few minor things that I've expected to change before merging this (see [this link](https://github.com/dooboolab/react-native-iap/blame/master/README.md#L38) for example).

I've done it like this for mine:
1. Checkout the master branch.
2. `cd docs`
3. `npm install`
4. Set `GIT_USER` and other variables depending on your setup (https://docusaurus.io/docs/deployment#environment-settings)
5. Run `npm run deploy`

Let me know if you need any help.

**Comment by hyochan** [comment]
@giovannicimolin Whoops. It is already `merged`. I hope you to re-submit a `PR`.

I am also setting up `CI` in #1375

**Comment by giovannicimolin** [comment]
@hyochan I'll submit a PR with the final fixes + change the link to `dooboolab`'s hosted version of the docs once it's deployed.

**Comment by hyochan** [comment]
@giovannicimolin Thanks. The website is currently deployed in react-native-iap.dooboolab.com.

**Comment by hyochan** [comment]
Please make sure of `rebasing` your repo before further updates @giovannicimolin

**Pull request merged**